### ISSUE TITLE
Allow setting crop in pixels instead of ratios

### ISSUE THREAD

**Issue by e00E** [baffler/OBS-OpenVR-Input-Plugin#34]
Currently the crop menu works internally as a float and in the obs UI can be incremented only up to 1% steps.
This makes exact cropping impossible or annoying because you first have to do math to see which float value corresponds to the pixel value you want and then put this exact float value into the ini file.

A better solution is to allow using both ratios or pixels in the ui. Depending on hard it is to support both at the same time, it would still be better to replace the current approach with pixels as that feels like the more natural unit.

I might be able to create a PR for this myself, but I'd like to know whether you would accept it first.

**Comment by baffler** [comment]
I do agree the cropping is a pain, I think using pixels would probably work out better. I had planned on using the Overlay View (current version is using the Mirror View). Using the overlay view would allow the cropping to be done by the SDK instead of having to set values yourself. As of right now it's working but pretty buggy. Anyways, I would accept the PR if it's going to easier to crop using pixels which I think it should be.

**Comment by e00E** [comment]
I implemented this but here are some problems:

* We don't know the device width and height until openvr is initialized. So if we want to use a slider in the UI it is unclear what the max value should be until then. Based on the plugin api it seems possible to update the max value at that point. What I like more is hiding the sliders until openvr is initialized since it doesn't make sense for users to edit them until then as they won't be able to see how their edit changes the view.
* When I quickly enter several numbers into the text box (like when typing `1234` with a rolling hand motion), `win_openvr_init` is only called once with the first edit (`1`) so it doesn't update to the correct final value. Not sure why this happens.

I also found a workaround that doesn't require code changes: It is possible to crop the source pixel accurately from within obs by right clicking it in the preview and editing the transform. This way I can do rough initial cropping with the ratio based ui and then do fine adjustments through the transform. This is a little more work than if this plugin supported it directly but it does work.
